hazelcast-vnet/azuredeploy.json

```diff
@@ -0,0 +1,230 @@
+      "type": "string",
+      "metadata": {
+        "description": "Password for hazelcast grid"
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "instanceCount": {
+      "type": "int",
+      "metadata": {
+        "description": "The number of hazelcast nodes in the grid"
+      }
+    },
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "The tag used to identify this cluster"
+      }
+    },
+    "clusterPort": {
+      "type": "string",
+      "metadata": {
+        "description": "The cluster port to identify this cluster"
+      }
+    },
+    "minHeapSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The minumum heap size for each hazelcast jvm"
+      }
+    },
+    "maxHeapSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The maximum heap size for each hazelcast jvm"
+      }
+    },
+    "aadClientId": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal client ID"
+      }
+    },
+    "aadClientSecret": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal client secret"
+      }
+    },
+    "aadTenantId": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal tenant ID"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The size of each instance VM"
+      },
+      "allowedValues": [
+        "Standard_DS1",
+        "Standard_DS2",
+        "Standard_DS3",
+        "Standard_DS4",
+        "Standard_DS8",
+        "Standard_DS16"
+      ]
+    },
+    "ubuntuOSVersion": {
+      "type": "string",
+      "defaultValue": "14.04.2-LTS",
+      "metadata": {
+        "description": "The Ubuntu version for deploying the Docker containers. This will pick a fully patched image of this given Ubuntu version. Allowed values: 14.04.2-LTS, 14.04-DAILY, 15.04"
+      },
+      "allowedValues": [
+        "14.04.2-LTS",
+        "14.04-DAILY",
+        "15.10"
+      ]
+    },
+    "templateBaseUrl": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/tree/master/hazelcast-vnet/",
+      "metadata": {
+        "description": "The base URL for the bootstrap files"
+      }
+    }
+  },
+  "variables": {
+    "apiVersion": "2015-01-01"
+  },
+  "resources": [{
+    "name": "hazelcast-vnet",
+    "type": "Microsoft.Resources/deployments",
+    "apiVersion": "[variables('apiVersion')]",
+    "properties": {
+      "mode": "[parameters('deploymentMode')]",
+      "templateLink": {
+        "uri": "[concat(parameters('templateBaseUrl'), 'azuredeploy-', parameters('vmAuthMethod'), '.json')]",
```

Comment:
Btw, you don't need multiple files to password or ssh auth. It can be done in the same template. See here https://github.com/Azure/azure-quickstart-templates/blob/master/checkpoint-single-nic/azuredeploy.json#L137 

Saves a lot of repeated code